plan9/srv9p/example/netshell/main.go

```diff
@@ -236,31 +239,30 @@ func newConsole(r io.Reader, w io.Writer) *console {
 // consctlClose handles a close of /dev/consctl; it turns raw mode off.
 func (c *console) consctlClose() {
 	c.mu.Lock()
+	defer c.mu.Unlock()
 	c.raw = false
-	c.mu.Unlock()
 }
 
 // consctlWrite handles writes to /dev/consctl.
 func (c *console) consctlWrite(data []byte) (int, error) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
-	s := string(data)
+	s := strings.TrimSpace(string(data))
 	switch s {
-	case "rawon", "rawon\n": // turn raw mode on
+	case "rawon": // turn raw mode on
 		c.raw = true
 		c.read = append(c.read, c.line...)
 		c.line = nil
 		return len(data), nil
 
-	case "rawoff", "rawoff\n": // turn raw mode off
+	case "rawoff": // turn raw mode off
 		c.raw = false
 		return len(data), nil
 	}
 
-	if strings.HasPrefix(s, "notepg") { // notepg n means send interrupt notes to /proc/n/notepg
-		s = strings.TrimSpace(strings.TrimPrefix(s, "notepg"))
-		if s == "" {
+	if s, ok := strings.CutPrefix(s, "notepg"); ok { // notepg n means send interrupt notes to /proc/n/notepg
+		if _, err := strconv.ParseUint(s, 10, 64); err != nil {
 			return 0, fmt.Errorf("bad notepg")
 		}
 		fd, err := syscall.Open("/proc/"+s+"/notepg", syscall.O_WRONLY)
```

Comment:
I was worried about malicious path traversal here, writing "notepg../../../../foo" to the ctl file for the pid.